$OpenBSD: patch-src_socket_c,v 1.4 2018/05/12 14:37:36 jca Exp $

Use functions provided by recent LibreSSL versions.

Fix format string.

Index: src/socket.c
--- src/socket.c.orig
+++ src/socket.c
@@ -40,7 +40,8 @@
 # include <openssl/ssl.h>
 # include <openssl/err.h>
 # include <openssl/x509v3.h>
-# if OPENSSL_VERSION_NUMBER < 0x10100000L
+# if OPENSSL_VERSION_NUMBER < 0x10100000L \
+	|| (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x2070100fL)
 #  define X509_OBJECT_get0_X509(o) ((o)->data.x509)
 #  define X509_STORE_get0_objects(o) ((o)->objs)
 # endif
@@ -470,7 +471,7 @@ socket_connect_one( conn_t *sock )
 		struct sockaddr_in6 *in6 = ((struct sockaddr_in6 *)ai->ai_addr);
 		char sockname[64];
 		in6->sin6_port = htons( sock->conf->port );
-		nfasprintf( &sock->name, "%s ([%s]:%hu)",
+		nfasprintf( &sock->name, "%s ([%s]:%d)",
 		            sock->conf->host, inet_ntop( AF_INET6, &in6->sin6_addr, sockname, sizeof(sockname) ), sock->conf->port );
 	} else
 #endif
@@ -482,7 +483,7 @@ socket_connect_one( conn_t *sock )
 		in->sin_addr.s_addr = *((int *)*sock->curr_addr);
 #endif
 		in->sin_port = htons( sock->conf->port );
-		nfasprintf( &sock->name, "%s (%s:%hu)",
+		nfasprintf( &sock->name, "%s (%s:%d)",
 		            sock->conf->host, inet_ntoa( in->sin_addr ), sock->conf->port );
 	}
 
